opensearch/sycamore-opensearch.sh

```diff
@@ -411,6 +375,123 @@ END
     echo "OPENAI_TASK_ID=${OPENAI_TASK_ID}" >>"${PERSISTENT_ENV_TMP}"
 }
 
+deploy_model_better() {
+    local model_id="$1" # if empty will be derived from task
+    local task_id="$2"
+    local name="$3"
+    [[ -z "${task_id}" ]] && die "ERROR missing task_id"
+    [[ -z "${name}" ]] && die "ERROR missing name"
+
+    if [[ -z "${model_id}" ]]; then
+        wait_task "${task_id}" "$name model to become ready"
+        model_id="${GET_TASK_MODEL_ID}"
+        echo "Fetched $name model id $model_id for task_id $task_id"
+    else
+        echo "Using existing model id ${model_id}"
+    fi
+
+    # Create deploy task
+    local deploy_model_log_file="${ARYN_STATUSDIR}/curl.better_deploy_model_task.${name}"
+    spawn_deploy_model_task "${name}" "${model_id}"
+    local deploy_task_id="$(jq -r '.task_id' "${deploy_model_log_file}")"
+
+    # Cycle on the task status
+    debug "Wait for deploy task to finish"
+    local deploy_status_file="${ARYN_STATUSDIR}/curl.deploy_task_status.${name}"
+    local deploy_task_search_file="${ARYN_STATUSDIR}/curl.deploy_task_search_result.${name}"
+    local i
+    local max_reps=40
+    for i in $(seq "${max_reps}"); do
+        _curl "${BASE_URL}/_plugins/_ml/tasks/${deploy_task_id}" -o "${deploy_status_file}"
+        local status="$(jq -r '.state' "${deploy_status_file}")"
+        debug "${status}"
+        # Case 1: RUNNING / state not found. Task is still running so wait.
+        if [[ "${status}" == 'null' || "${status}" == 'RUNNING' || "${status}" == 'CREATED' ]]; then
+            echo "Waiting for ${name} to deploy... ${i}/${max_reps}"
+            sleep 1
+        # Case 2: COMPLETED. Task is completed, so exit
+        elif [[ "${status}" == 'COMPLETED' ]]; then
+            echo "Deployed ${name} successfully"
+            MODEL_ID="${model_id}"
+            return 0
+        # Case 3: FAILED. Handle Memory Circuit Breaker error by waiting a bit but otherwise fail out
+        elif [[ "${status}" == 'FAILED' ]]; then
+            # handle error
+            debug "Deploy task failed for ${name}"
+            local error="$(jq -r '.error' "${deploy_status_file}")"
+            debug "${error}"
+            local worker_node="$(jq -r '.worker_node[0]' "${deploy_status_file}")"
+            local error_message="$(echo "${error}" | jq -r ".${worker_node}")"
+            debug "${error_message}"
+            echo "Deploy task failed for ${name}: ${error_message}"
+            # Memory Circuit Breaker error: wait 20 seconds and the try to deploy again
+            if [[ "${error_message}" = "Memory Circuit Breaker is open, please check your resources!" ]]; then
+                local wait_time=20
+                echo "Waiting for ${wait_time}s while GC runs before retrying deploy"
+                for j in $(seq "${wait_time}"); do
+                    echo -n "."
```

Comment:
This works poorly with docker compose, it forces newlines.
But since you're not checking to see if it's fixed along the way, you might as well just do:
sleep 20.

Is there a reason to not do a shorter sleep and retry from the top? If you set max reps to 60 and sleep 5 here, then it would wait 300s in the face of circuit breaker issues.  The advantage of the shorter sleep is less waiting to get started.